jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/DMatrix.scala

```diff
@@ -70,15 +70,33 @@ class DMatrix private[scala](private[scala] val jDMatrix: JDMatrix) {
   @throws(classOf[XGBoostError])
   def this(headers: Array[Long], indices: Array[Int], data: Array[Float], st: JDMatrix.SparseType,
            shapeParam: Int) {
-    this(new JDMatrix(headers, indices, data, st, shapeParam))
+    this(new JDMatrix(headers, indices, data, st, shapeParam, Float.NaN, -1))
```

Comment:
Noting that we are changing the behavior of `missing` in the JVM package. This may interact badly with `VectorAssembler` of Spark, which treats 0 as missing. The XGBoost doc has a dedicated section about this behavior: https://xgboost.readthedocs.io/en/stable/jvm/xgboost4j_spark_tutorial.html#dealing-with-missing-values.

* Make sure to update the Spark tutorial.
* Let's add tests with VectorAssembler and ensure that it behaves reasonably.

cc @wbo4958

Reply:
This PR doesn't change the existing behavior as the default value for `missing` is Float.NaN.

Reply:
How about this line? It's setting `missing` to 0:
https://github.com/dmlc/xgboost/blob/8af98e30fccbe03402720a4d0143bb253080c883/jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java#L134

Reply:
Actually never mind, we are replacing `XGDMatrixCreateFromCSREx`